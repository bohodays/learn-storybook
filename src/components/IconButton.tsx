interface IIconButtonProps {
  alt: string;
  iconPath: string;
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

const IconButton = ({ iconPath, alt, onClick }: IIconButtonProps) => {
  return (
    <button onClick={onClick}>
      <img src={iconPath} alt={alt} />
    </button>
  );
};

export default IconButton;
