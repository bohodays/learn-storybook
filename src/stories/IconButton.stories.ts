import type { Meta, StoryObj } from "@storybook/react-vite";

import { fn } from "storybook/test";
import IconButton from "../components/IconButton";

// Docs 정보
const meta = {
  title: "Buttons/IconButton", // 경로를 나타냄
  component: IconButton,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  argTypes: {
    alt: { control: "text", desciption: "이미지의 alt 속성", defaultValue: "" },
    iconPath: {
      control: "text",
      description: "이미지의 경로",
      defaultValue: "",
    },
    onClick: { action: "clicked", description: "버튼 클릭 이벤트" },
  },
} satisfies Meta<typeof IconButton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    alt: "icon",
    // iconPath: deleteIcon,
    iconPath: "https://kr.object.ncloudstorage.com/icons/ic-delete-dark.svg",
    onClick: () => {},
  },
};
