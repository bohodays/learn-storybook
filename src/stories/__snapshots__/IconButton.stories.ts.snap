// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons/IconButton Default smoke-test 1`] = `
<button>
  <img alt="icon"
       src="https://kr.object.ncloudstorage.com/icons/ic-delete-dark.svg"
  >
</button>
`;
